libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/GuidanceViewImageProvider.kt

```diff
@@ -25,12 +28,7 @@ class GuidanceViewImageProvider() {
         private val USER_AGENT_VALUE = "MapboxJava/"
     }
 
-    /**
-     * This is added a solution to bypass the issue with Picasso where [com.squareup.picasso.Target]
-     * get's garbage collected and guidance view is hence not rendered. Found the solution here
-     * https://stackoverflow.com/questions/24180805/onbitmaploaded-of-target-object-not-called-on-first-load#answers
-     */
-    private val targets: MutableList<Target> = mutableListOf()
+    private val coroutineScope = CoroutineScope(Dispatchers.IO)
```

Comment:
Is there some reason you're not using ThreadController?

Reply:
please use `ThreadController ` to handle all coroutines from a single place